assets/src/design-system/components/contextMenu/menu.js

```diff
@@ -35,93 +35,134 @@ const SEPARATOR_BOTTOM_CLASS = 'separatorBottom';
 
 const MenuWrapper = styled.div(
   ({ theme }) => css`
-    padding: 5px 0 0;
+    padding: 0;
     background-color: ${theme.colors.bg.primary};
     pointer-events: none;
     border-radius: ${theme.borders.radius.small};
     border: 1px solid ${theme.colors.border.disable};
-    width: 200px;
+    width: ${({ isIconMenu }) => (isIconMenu ? '40px' : '200px')};
   `
 );
+MenuWrapper.propTypes = {
+  isIconMenu: PropTypes.bool,
+};
 
-const MenuList = styled.ul(
-  ({ theme }) => css`
-    background-color: ${theme.colors.bg.primary};
-    border-radius: ${theme.borders.radius.small};
-    margin: 0;
-    padding: 4px 0;
-    pointer-events: auto;
-    list-style: none;
+const separatorCSS = css`
+  display: block;
+  content: '';
+  width: 100%;
+  height: 1px;
+  background-color: ${({ theme }) => theme.colors.bg.tertiary};
+  ${({ isIconMenu, theme }) =>
+    isIconMenu &&
+    css`
+      width: 40%;
+      margin: auto;
+      background-color: ${theme.colors.fg.primary};
+    `}
+`;
+separatorCSS.propTypes = {
+  isIconMenu: PropTypes.bool,
+};
 
-    a {
-      background-color: transparent;
-      text-decoration: none;
+const MenuList = styled.ul`
+  background-color: ${({ theme }) => theme.colors.bg.primary};
+  border-radius: ${({ theme }) => theme.borders.radius.small};
+  margin: 0;
+  padding: ${({ isIconMenu }) => (isIconMenu ? 0 : '4px 0')};
+  pointer-events: auto;
+  list-style: none;
+
+  a {
+    background-color: transparent;
+    text-decoration: none;
+  }
+
+  li {
+    a,
+    button,
+    div {
+      display: flex;
+      justify-content: space-between;
+      align-items: center;
+      padding: 8px 16px;
+      border: 0;
+      transition: background-color ${BUTTON_TRANSITION_TIMING};
     }
+    ${({ isIconMenu }) =>
+      isIconMenu &&
+      css`
+        margin: 0;
+        button {
+          padding: 0;
+          margin: 4px auto;
+        }
+        div,
+        a {
+          padding: 0;
+        }
+      `}
```

Comment:
all new.  Need to separate button from div/a because of tooltips on the icon menu.